core/services/synchronization/telemetry_ingress_client.go

```diff
@@ -58,9 +57,9 @@ type telemetryIngressClient struct {
 }
 
 type TelemPayload struct {
-	Ctx             context.Context
-	Telemetry       []byte
-	ContractAddress common.Address
+	Ctx        context.Context
+	Telemetry  []byte
+	ContractID string
```

Comment:
I wonder if it would be possible to put this in the relay config instead... but then you'd lose the unique guarantee at DB level, right?

Reply:
Perhaps we could add functional unique constraints on the relay config JSON instead... just thinking out loud here

Reply:
That could work, we should use an index on that `JSONB` subfield (`config->>'contractID'`)

Reply:
the contractID is common to all OCR2 relays though, I'm not sure the value of putting it in the relay config for all relays? Can't get db level type safety either way

Reply:
On balance I think keeping it at the top level is ok